inc/entity.class.php

```diff
@@ -3415,7 +3457,65 @@ static function getIcon() {
       return "fas fa-layer-group";
    }
 
+   /**
+    * Get values for contracts_id_default field
+    *
+    * @since 9.5
+    *
+    * @return array
+    */
+   public static function getDefaultContractValues($entities_id): array {
+      $values = [
+         self::CONFIG_PARENT => __('Inheritance of the parent entity'),
+         -1 => __('Contract in ticket entity'),
```

Comment:
Can you confirm that this will result in usage of first found not expired contract from the entity of the newly created ticket ?

Reply:
Yes.

Reply:
I usually do not like this kind of logic, as it may produce unpredictable result in many situations.

If we keep this possibility, label should indicate this unpredictable state:
```suggestion
         -1 => __('First found valid contract in ticket entity'),
```

@orthagh @trasher  ?